Cargo: making requisitions form useful

Aldimensch opened the pull request:
## About The Pull Request

Only paperwork is touched in \code\modules\cargo\orderconsole.dm

## Why It's Good For The Game

It makes the Cargo requisition form useful. Cargonian will be happy!

<a href='https://postimg.cc/LnBQqqY8' target='_blank'><img src='https://i.postimg.cc/LnBQqqY8/fulp-new-cargo.jpg' border='0' alt='fulp-new-cargo'/></a>

## Changelog

🆑
qol: As a player, you can now see more information on the cargo requisition forms.
/🆑

Aldimensch committed:
none

Aldimensch committed:
Adding requisition informations

Adding more information on the requisition form, making it useful again.

Aldimensch committed:
fixed small visual error

<a href='https://postimg.cc/LnBQqqY8' target='_blank'><img src='https://i.postimg.cc/LnBQqqY8/fulp-new-cargo.jpg' border='0' alt='fulp-new-cargo'/></a>

Aldimensch committed:
fixed un-intended change

JohnFulpWillard commented:
Please add a changelog since you got player-facing changes

JohnFulpWillard commented on code/modules/cargo/orderconsole.dm:
```diff
@@ -322,21 +346,6 @@
 				usr.investigate_log("called the supply shuttle.", INVESTIGATE_CARGO)
 				say("The supply shuttle has been called and will arrive in [SSshuttle.supply.timeLeft(600)] minutes.")
 				SSshuttle.moveShuttle(cargo_shuttle, docking_home, TRUE)
-			if(!length(cart_list))
```
Now there'll have a paper even if nothing was purchased, was this intentional?

Aldimensch replied:
No, fixed the issue. No paper will be generated when nothing is in the shopping_list.

JohnFulpWillard commented on code/modules/cargo/orderconsole.dm:
```diff
@@ -313,6 +313,30 @@
+			requisition_text += "<hr/>"
+			requisition_text += "Time of Order: [station_time_timestamp()]<br/><br/>"
+
+			for(var/datum/supply_order/order in SSshuttle.shopping_list)
```
What's the difference between this and 'cart_list'?

JohnFulpWillard reviewed (commented):
I like this change, minor comments though

Aldimensch committed:
Changes based on comments

Removed last code fragments which are no longer used; cart_list
When nothing is in shopping_ist no paper will be generated. Reduces paper consumption of NT holdings!

Fikou commented on code/modules/cargo/orderconsole.dm:
```diff
@@ -302,16 +302,29 @@
+					requisition_text += "- Ordered by: [order.orderer] ([order.orderer_rank])</br>"
+					var/paying_account = "[order.paying_account]"
+					if(paying_account)
+						requisition_text += "- Paid Privatly by: [order.paying_account.account_holder]<br/>"
```
```suggestion
						requisition_text += "- Paid Privately by: [order.paying_account.account_holder]<br/>"
```

Fikou commented on code/modules/cargo/orderconsole.dm:
```diff
@@ -302,16 +302,29 @@
+						requisition_text += "- Paid Privatly by: [order.paying_account.account_holder]<br/>"
+					var/reason = "[order.reason]"
+					if(reason != "")
+						requisition_text += "- Reason given: [reason]</br>"
```
```suggestion
						requisition_text += "- Reason Given: [reason]</br>"
```

Fikou commented on code/modules/cargo/orderconsole.dm:
```diff
@@ -302,16 +302,29 @@
+					requisition_text += "<b>[order.pack.name]</b></br>"
+					requisition_text += "- Order ID: [order.id]</br>"
+					var/restrictions = "[SSid_access.get_access_desc(order.pack.access)]"
+					if(restrictions != "")
```
you can just check if(restrictions)

Fikou commented on code/modules/cargo/orderconsole.dm:
```diff
@@ -302,16 +302,29 @@
+					if(paying_account)
+						requisition_text += "- Paid Privatly by: [order.paying_account.account_holder]<br/>"
+					var/reason = "[order.reason]"
+					if(reason != "")
```
can just check if(reason)

Fikou commented on code/modules/cargo/orderconsole.dm:
```diff
@@ -302,16 +302,29 @@
+					var/paying_account = "[order.paying_account]"
+					if(paying_account)
+						requisition_text += "- Paid Privatly by: [order.paying_account.account_holder]<br/>"
+					var/reason = "[order.reason]"
```
can't you just do reason = order.reason? same with restrictions

Aldimensch replied:
I can try it. Since I have no idea how the compiler handles String conversions nor how it handles String If-Statments. It just was the laziest and easiest way to handle the If-Statement, without the knowledge of how exactly Datums and Variables are handled by the compiler and engine.

Thanks for the valuable pointers ;)

JohnFulpWillard replied:
I think it's fine this way

Aldimensch committed:
Fixed typo

Co-authored-by: Fikou <23585223+Fikou@users.noreply.github.com>

Aldimensch commented on code/modules/cargo/orderconsole.dm:
```diff
@@ -302,16 +302,29 @@
+				var/requisition_text = "<h2>[station_name()] Supply Requisition</h2>"
+				requisition_text += "<hr/>"
+				requisition_text += "Time of Order: [station_time_timestamp()]<br/><br/>"
+				for(var/datum/supply_order/order in SSshuttle.shopping_list)
```
What exactly does "as anything" and how does it compare to "in"?

joksulainen commented:
this is a much needed change after whoever changed it to that terribleness with no important info anywhere in sight on the paper

Aldimensch committed:
Fikou comment based changes

Improved code, fewer string comparisons, and removed datum comparison in the 'for' loop.
Fixed grammatic errors

Aldimensch committed:
Merge branch 'tgstation:master' into cargo-making-requisitions-form-useful

Aldimensch commented:
I thank @JohnFulpWillard and @Fikou for the tips and corrections given.
As this is my first Byond code ever written and published, I am very glad this went so smoothly.

All hail Cargonia!
